pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -564,16 +564,16 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 		return false
 	}
 
-	// Allocates from reusableDevices list first.
-	if allocateRemainingFrom(reusableDevices) {
-		return allocated, nil
-	}
-
 	// Needs to allocate additional devices.
 	if m.allocatedDevices[resource] == nil {
 		m.allocatedDevices[resource] = sets.NewString()
 	}
```

Comment:
Would it make sense to move this check up the stack to prevent crashes in other places? Ideally it should be done as soon as the resource value is known.